include/gnc_containers.hpp

```diff
@@ -0,0 +1,60 @@
+//
+// include/gnc_containers.hpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#ifndef PAN_PSIM_INCLUDE_GNC_CONTAINERS_HPP_
+#define PAN_PSIM_INCLUDE_GNC_CONTAINERS_HPP_
+
+#include <array>
+#include <type_traits>
+
+namespace gnc {
+
+/** @class CircularBuffer
+ *  @tparam T Data type stored in the buffer.
+ *  @tparam N Maximum number of elements that can be stored in the buffer.
+ *  Simple circular buffer implementation that requires the data type to have a
+ *  default constructor. */
+template <typename T, std::size_t N>
+class CircularBuffer {
+ public:
+  constexpr CircularBuffer();
+  constexpr CircularBuffer(CircularBuffer<T, N> const &) = default;
+  constexpr CircularBuffer(CircularBuffer<T, N> &&) = default;
+  constexpr CircularBuffer<T, N> &operator=(CircularBuffer<T, N> const &) = default;
+  constexpr CircularBuffer<T, N> &operator=(CircularBuffer<T, N> &&) = default;
+  /** @returns True if the buffer contains the maximum number of elements. */
+  constexpr bool full() const;
+  /** @returns True if the buffer contains no elements. */
+  constexpr bool empty() const;
+  /** @returns Number of elements currently in the buffer. */
+  constexpr std::size_t size() const;
```

Comment:
size_t is available in `<cstddef>` without the need for std::

Reply:
Didn't know that was a thing - although forgot I didn't use `array` because of `constexpr` issues so I gotta change some of the include stuff up anyway. Seems a little weird that you don't need `std::` though if it's part of the STL. That `size_t` in particular is actually probably from `C` and we may want to avoid using it for consistency in `cxx` code. I've been having issues including both `math.h` and `cmath` due to naming conflicts and such.

Reply:
`size_t` is enough of a primitive type that I just went for using it in Flight Software; it's all over the place.` <cstddef>` wouldn't interfere with math, I think, so it's available if you want to make the code more consistent with FSW.